### PR TITLE
Algolia search - Full Implementation

### PR DESCRIPTION
**Pull request by JMStewart00**
## Summary / Approach

- Add the search_api_algolia module
- Set up webpack and a custom module to build a react app
- Disable search api views and delete the blocks
- Some other odds and ends stuff. 

## Testing Instruction(s)

In the multidev env use the new (to-be-created manually) search page. 
use the flyout search tool. 

## Screenshots

Lexington_ Kentucky_ Horse Capital of the World_ (2)
Search _ City of Lexington

## Metadata

| Question | Answer |
|----------|--------|
| Did you perform a self-review of this PR? | some
| Documentation reflects changes? | not yet
| Unit/Functional tests reflect changes? | n/a
| Did you perform browser testing? | yeah FF, Chrome
| Did you provide detail in the summary on how and where to test this branch? | yes
| Relevant links | no links, POC agreement with Nick

### CODE REVIEW

**Comment by TaylorAkin** [comment]
[Visit Site](https://pr-467-lexky-d8.pantheonsite.io/)

Created multidev environment [pr-467](https://dashboard.pantheon.io/sites/085efc9d-6f4f-4d02-a80b-9d5cf086a982#pr-467) for lexky-d8.

File: config/sync/core.extension.yml [hunk 1]
```diff
@@ -38,7 +39,6 @@ module:
   dropzonejs_eb_widget: 0
   dynamic_page_cache: 0
   editor: 0
-  editor_email_link: 0
```
Comment (JMStewart00) [inline config/sync/core.extension.yml:41]: I think this needs to be added back in, correct @TaylorAkin ?

File: config/sync/search_api.server.algolia.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+dependencies:
+  module:
+    - search_api_algolia
+id: algolia
+name: Algolia
+description: ''
+backend: search_api_algolia
+backend_config:
+  application_id: U49JR38RZ5
+  api_key: redacted
```
Comment (JMStewart00) [inline config/sync/search_api.server.algolia.yml:13]: This is `redacted` but added to the config_ignore settings to make sure it doesn't disconnect.

**Review by chris-allen — APPROVED** [review]
This looks slick. My biggest question probably would've been cleared up by documentation:

What's the intended setup for using algolia locally in lando? I assume you can use all the same `sapi-*` drush commands but do we need to setup our own index (not sure what algolia calls it) for local development?

**Comment by JMStewart00** [comment]
> This looks slick. My biggest question probably would've been cleared up by documentation:
> 
> What's the intended setup for using algolia locally in lando? I assume you can use all the same `sapi-*` drush commands but do we need to setup our own index (not sure what algolia calls it) for local development?

Ideally we use config split and set up an index just for local development and testing. We could have indexes for every environment. 

I still need to configure that and write some documentation. In a perfect world we could do something here with the module that we reuse as a starter.

File: config/sync/core.extension.yml (hunk 1 above)
Comment (TaylorAkin) [reply config/sync/core.extension.yml:41]: Yea it does

File: web/modules/custom/apax_search_algolia/react/src/components/Autocomplete.js [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+
+    const { onTouchStart, onTouchMove, onMouseDown } = getEnvironmentProps({
+      formElement: formRef.current,
+      inputElement: inputRef.current,
+      panelElement: panelRef.current,
+    });
+
+    window.addEventListener("mousedown", onMouseDown);
+    window.addEventListener("touchstart", onTouchStart);
+    // window.addEventListener("touchmove", onTouchMove);
```
Comment (TaylorAkin) [inline web/modules/custom/apax_search_algolia/react/src/components/Autocomplete.js:77]: commented out line

File: web/modules/custom/apax_search_algolia/react/src/components/Autocomplete.js [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+
+    return () => {
+      window.removeEventListener("mousedown", onMouseDown);
+      window.removeEventListener("touchstart", onTouchStart);
+      // window.removeEventListener("touchmove", onTouchMove);
```
Comment (TaylorAkin) [inline web/modules/custom/apax_search_algolia/react/src/components/Autocomplete.js:82]: commented out line

File: web/modules/custom/apax_search_algolia/react/src/utils/constants.jsx [hunk 5]
```diff
@@ -0,0 +1,6 @@
+export const DEBOUNCE_TIME = 300;
+export const LARGE_SCREEN_SIZE = 992; // matches scss variable for lg breakpoint
+
+// Algolia index IDs, from IDs in dashboard
+export const INDEX_ID = 'prod_lexky';
```
Comment (TaylorAkin) [inline web/modules/custom/apax_search_algolia/react/src/utils/constants.jsx:5]: What's going on here? @JMStewart00

**Review by TaylorAkin — APPROVED** [review]

File: web/modules/custom/apax_search_algolia/react/src/utils/constants.jsx (hunk 5 above)
Comment (JMStewart00) [reply web/modules/custom/apax_search_algolia/react/src/utils/constants.jsx:5]: This is the id of the index as it is in Algolia. React instantsearch needs to know where to pull the information from.

**Pull request merged**